### PR TITLE
#2294 Use agreed set of v2 statuses

### LINKED ISSUE #2294: Use agreed set of v2 statuses
Following review of version 2 Stages & Statuses there are some updates to make to the final set of v2 statuses, as follows:

- [x] Remove new statuses as specified on [spreadsheet](https://docs.google.com/spreadsheets/d/1BjQ6dtrGRWZSQWVPCwB0Q5exiDjUCEIK/edit?usp=sharing&ouid=106025876472425620100&rtpof=true&sd=true)
- [x] Rename display value of new statuses as per column **_v2 Amended Display Value_**
- [x] Relocate Character Rejected status in process according to spreadsheet
- [x] Add new status as per row 23 on spreadsheet
- [x] Remove tasks related to statuses that are going to be removed
- [x] Support statuses which move the application back a stage
- [x] Application list should show processing status rather than application status
- [x] Override tab titles for Diversity and Outreach reports

See this ticket for more detail and a link to the spreadsheet: jac-uk/digital-platform#1002

See [Candidate process flow](https://miro.com/app/board/uXjVN7KsLv8=/)

_Note we will need to update digital-platform as well as Admin_

---

Following review on 23/02/2024 the stages have been amended and the statuses changed.
 - [x] Update code to use latest stages
 - [ ] Update code to use latest statuses

### PR DESCRIPTION
**Pull request by warrensearle**
## What's included?

Following review of version 2 Stages & Statuses there are some updates to make to the final set of v2 statuses, as follows:

- [x] Remove new statuses as specified on [spreadsheet](https://docs.google.com/spreadsheets/d/1BjQ6dtrGRWZSQWVPCwB0Q5exiDjUCEIK/edit?usp=sharing&ouid=106025876472425620100&rtpof=true&sd=true)
- [x] Rename display value of new statuses as per column **_v2 Amended Display Value_**
- [x] Relocate Character Rejected status in process according to spreadsheet
- [x] Add new status as per row 23 on spreadsheet
- [x] Remove tasks related to statuses that are going to be removed
- [x] Support statuses which move the application back a stage
- [x] Application list should show processing status rather than application status
- [x] Override tab titles for Diversity and Outreach reports

See this ticket for more detail and a link to the spreadsheet: jac-uk/digital-platform#1002

See [Candidate process flow](https://miro.com/app/board/uXjVN7KsLv8=/)

Closes #2294 

## Who should test?
✅ Product owner
✅ Developers

## How to test?
Example exercise: [2294 Stages and statuses](https://jac-admin-develop--pr2325-feature-2294-l1fi4273.web.app/exercise/EzikIl5VGsMblAcCCCu1/stages/applied), [Stages and statuses test1](https://jac-admin-develop--pr2325-feature-2294-l1fi4273.web.app/exercise/Oy4hPcHarUc0CMQau7Hh/dashboard)

- Navigate to Stages
- For each of the stages select an application then click the Set status button
- On the Update page check that the correct statuses are displayed

(refer to spreadsheet mentioned above)

## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

## Additional context

**Applied**
image

**Shortlisted**
image

**Selectable**
image

**Recommended**
image

## Related permissions
Have permissions been considered for this functionality?
- No permission changes required
- Permissions have been added / updated. Details:

---
PREVIEW:PRODUCTION
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

**Review by HalcyonJAC — APPROVED** [review]
Looks good to me

**Review by tomlovesgithub — APPROVED** [review]

**Comment by warrensearle** [comment]
This now includes a basic configuration form to set the application form version and the processing version. For processing version changes this additionally calls `updateApplicationRecordStageStatus` which migrates data between the different versions

**Comment by warrensearle** [comment]
NOTE: this doesn't currently fix the reports (e.g. Diversity Report) to ensure they display the correct stages. Lets get the migration and Stages & Status pages correct first!

File: src/views/Exercise/Configuration/ProcessingVersion.vue [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          value="2"
+          label="Processing version 2.0"
+          hint="Introduced in March 2024, processing version 2.0 updates stages and statuses and includes a number of new tasks"
+        />
+        <RadioItem
+          value="1"
+          label="Processing version 1.0"
+          hint="The original!"
+          disabled
+        />
```
Comment (HalcyonJAC) [inline src/views/Exercise/Configuration/ProcessingVersion.vue:28]: It seems that the field type of `_processingVersion` is number. Should we change `value="2"` to `:value="2"`?

**Comment by HalcyonJAC** [comment]
@warrensearle I have deployed the cloud function to develop and also fixed the count of applicationRecords.

**Review by nickaddy — APPROVED** [review]

**Pull request merged**